worker/worker-execution/score-worker-execution-impl/src/test/java/io/cloudslang/worker/execution/services/ExecutionServiceTest.java

```diff
@@ -159,16 +166,50 @@ public void handleCancelledFlowsTest() {
 
 		boolean result = executionService.handleCancelledFlow(exe);
 
-		Assert.assertEquals(exe.getPosition(), null);
-		Assert.assertEquals(result, true);
+		assertNull(exe.getPosition());
+		assertTrue(result);
 
 		exe = new Execution(EXECUTION_ID_2,0L, 0L, new HashMap<String,String>(), null);
 
 		result = executionService.handleCancelledFlow(exe);
 
-		Assert.assertEquals(exe.getPosition(), null);
-		Assert.assertEquals(result, true);
+        assertNull(exe.getPosition());
+        assertTrue(result);
+
+	}
+
+	@Test
+	// branch is running and execution reaches sequential operation -> branch should be paused
+	public void handlePausedFlow_sequentialOperationReached() throws InterruptedException {
+
+		ExecutionStep executionStep = new ExecutionStep(EXECUTION_STEP_1_ID);
```

Comment:
(optional) The test is hard to read because of all these pojo creation + setters. You could hide for example the actionData in a method createActionData that also populates that map.